Userland/Libraries/LibGfx/PNGLoader.cpp

```diff
@@ -1072,7 +1061,11 @@ ErrorOr<ImageFrameDescriptor> PNGImageDecoderPlugin::frame(size_t index)
 
     if (m_context->state < PNGLoadingContext::State::BitmapDecoded) {
         // NOTE: This forces the chunk decoding to happen.
-        TRY(decode_png_bitmap(*m_context));
+        auto maybe_error = decode_png_bitmap(*m_context);
+        if (maybe_error.is_error()) {
```

Comment:
We wouldn't need this here if all the places that set error before would still set error. And you could still move from returning bools to returning errors. I.e. do

```
    if (!result.has_value()) {
        context.state = PNGLoadingContext::State::Error;
        return Error::from_string_literal("PNGImageDecoderPlugin: Decompression failed");
    }
```

(and so on)

and then just keep this here as it was.

Reply:
WDYT about this style https://github.com/SerenityOS/serenity/pull/17603 ?

Reply:
(FWIW I no longer like that style since it doesn't catch errors from code in libraries called by the decoder code. I redid it in ddc2cc886b588a0b5e4c742c5597983745c34ed7)